### PR TITLE
New Simple Interactable Cutscene Script + Example cutscene using it on Magitech Tile 1

### PR DESCRIPTION
**Pull request by ClemonsXLogana**
The point of this script is to allow me to create a "cutscenes" where two or more npcs are talking to each other. I put cutscene in quotations because I'm trying to treat it as much like normal dialogue as possible, by letting the player skip through the typing and the waiting in between lines being said so impatient players can get through it quicker. The cutscene will still play out at its normal time even if player leaves the trigger though, which I think is a nice middleground, making it seem like things are happening in the world between other characters, not just between one npc and the player, while still giving the player functional control to skip through or ignore.

I made this because there are simply too many different moments where I think it's important that npcs are actually talking to each other, rather than the only thing we can do currently which is just they talk to you and only you when you come near and that's it.

I tried my very best not to touch other NPC/dialogue scripts, but I did have to do it in one place, because I couldn't find a way to skip an NPC's current dialogue line without having the dialoguecontext.

I also changed the on trigger enter script on npcs cause i needed a way to retrigger any on trigger events when the cutscene ends, if player happens to still be in the trigger, so they dont have to walk out and back in to talk. I did leave a gap of 2 seconds in between though cause in my experience it was awkward for them to switch back to normal talking immediately.

Check out the Sample cutscene I created of the magitech initial portal open on Tile 1 to see how it works.

### CODE REVIEW

File: Slider/Assets/Scripts/Utility/Control/OnTriggerEnter.cs [hunk 1]
```diff
@@ -5,20 +5,46 @@
 
 public class OnTriggerEnter : MonoBehaviour
 {
-    public bool onPlayerEnter = true;
```
Comment (ChaseOB) [inline Slider/Assets/Scripts/Utility/Control/OnTriggerEnter.cs:8]: this is going to break a lot of shit

File: Slider/Assets/Scripts/Misc/SimpleInteractableCutscene.cs [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+        yield return new WaitWhile(() => character.IsTypingDialogue());
+
+        currentlyTyping = false;
+        currentlyWaitingAfterTyped = true;
+
+        float startTime = Time.time;
+        float currentTime = Time.time;
+        while ((currentTime - startTime < timeWaitAfterFinishedTyping) && !skipWaitingAfterTyped)
+        {
+            yield return new WaitForFixedUpdate();
```
Comment (randomerz) [inline Slider/Assets/Scripts/Misc/SimpleInteractableCutscene.cs:187]: do `yield return null;` here instead to wait for a normal frame (fixed update is a physics frame)

File: Slider/Assets/Scripts/Misc/SimpleInteractableCutscene.cs [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+    }
+
+    private void StartCutScene()
+    {
+        foreach (NPC character in cutsceneCharacters)
+        {
+            if (character.IsDialogueBoxActive())
+            {
+                character.DeactivateDialogueBox();
+            }
```
Comment (randomerz) [inline Slider/Assets/Scripts/Misc/SimpleInteractableCutscene.cs:115]: i'm not sure how reliable this method is to be honest so its prob a good thing to keep in mind for the future

Comment (ClemonsXLogana) [reply Slider/Assets/Scripts/Misc/SimpleInteractableCutscene.cs:115]: yeah i wasn't sure either, i looked around a bunch to see if there was a more definitely method but couldnt find one

**Pull request merged**